tests/test_client.py

```diff
@@ -26,46 +25,32 @@ def test_init(api_client: ApiClient):
     assert api_client
 
 
-def test_timeout_default(httpx_mock: HTTPXMock):
-    httpx_mock.add_response(
-        match_extensions={
-            "timeout": {
-                "connect": DEFAULT_TIMEOUT_CONFIG.connect,
-                "read": DEFAULT_TIMEOUT_CONFIG.read,
-                "write": DEFAULT_TIMEOUT_CONFIG.write,
-                "pool": DEFAULT_TIMEOUT_CONFIG.pool,
-            }
-        },
-        status_code=httpx.codes.BAD_GATEWAY,
+def test_base_url_override(respx_mock: respx.mock):
+    respx_mock.get(re.compile(r"^https://test/repos/foo/bar/import/issues")).mock(
+        return_value=httpx.Response(
+            httpx.codes.OK, text=get_fixture("response-multiple-check-status-of-multiple-issues.json")
+        )
+    )
+    ApiClient(http_client=HttpClient(token=GITHUB_TOKEN, base_url="https://test")).get_status_multiple(
+        "foo", "bar", datetime.now(tz=UTC)
     )
-    with pytest.raises(httpx.HTTPStatusError):
-        ApiClient(http_client=HttpClient(token=GITHUB_TOKEN)).get_status_multiple("foo", "bar", datetime.now(tz=UTC))
 
 
-def test_timeout_set(httpx_mock: HTTPXMock):
-    httpx_mock.add_response(
-        match_extensions={"timeout": {"connect": 123, "read": 123, "write": 123, "pool": 123}},
-        status_code=httpx.codes.BAD_GATEWAY,
-    )
+def test_headers_override(respx_mock: respx.mock):
+    respx_mock.route(headers={"Foo": "Bar"}).mock(return_value=httpx.Response(httpx.codes.BAD_GATEWAY))
     with pytest.raises(httpx.HTTPStatusError):
```

Comment:
You're mocking response to return bad gateway status code and expecting to raise `httpx.HTTPStatusError`. But in the same situation on the line 74 you're expecting to raise `httpx.HTTPError`. I wouldn't rely on the expected exception at all. You're trying to test headers override. Just use the same response from `test_base_url_override` but with additional headers. This is cleaner from my point of view.

Reply:
Good point, I was lazy and conflated several issues. I have unraveled them now. Thanks for your in-depth comments! Always motivates me to go the extra mile. It's hard to stay pedantic when you're alone.